aten/src/ATen/cudnn/Handle.cpp

```diff
@@ -17,8 +15,25 @@ struct Handle {
   Handle() : handle(NULL) {
     AT_CUDNN_CHECK(cudnnCreate(&handle));
   }
+  // std::vector.emplace() and push_back() may route through temporaries and call
+  // copy/move constructors along the way.  If this is the case, we don't want
+  // the destructors of temporaries to call cudnnDestroy on the handle.
+  // We can achieve safety (for the narrow case of stashing within STL containers)
+  // by defining copy and move constructors that transfer cudnnDestroy
+  // responsibility to the latest constructed object.  This is NOT a substitute for
+  // full-blown reference counting, but reference counting may be overkill here.
+  // Another alternative is to wrap the saved Handles in unique_ptrs, i.e.,
+  // unordered_map<int, vector<unique_ptr<Handle>>> created_handles;
+  void transfer(Handle& rhs) {
+    handle = rhs.handle;
+    rhs.handle = NULL;
+  }
+  Handle(Handle& rhs) { transfer(rhs); }
```

Comment:
This is not a thing. You can have copy constructor (`const Handle&`) or move constructor (`Handle&&`); a `Handle&` constructor is not a thing.